src/FunFair.CodeAnalysis.Tests/ProhibitedClassesDiagnosticsAnalyzerTests.cs

```diff
@@ -0,0 +1,159 @@
+using System.Threading.Tasks;
+using FunFair.CodeAnalysis.Tests.Helpers;
+using FunFair.CodeAnalysis.Tests.Verifiers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Xunit;
+
+namespace FunFair.CodeAnalysis.Tests
+{
+    public sealed class ProhibitedClassesDiagnosticsAnalyzerTests : CodeFixVerifier
+    {
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new ProhibitedClassesDiagnosticsAnalyzer();
+        }
+
+        [Fact]
+        public Task AssertFalseWithMessageIsAllowedAsync()
+        {
+            const string test = @"
+     using NonBlocking;
+
+     namespace ConsoleApplication1
+     {
+         class TypeName
+         {
+             void Test()
+             {
+                 ConcurrentDictionary<int,int> dictionary = new ConcurrentDictionary<int,int>();
+             }
+         }
+     }";
+
+            return this.VerifyCSharpDiagnosticAsync(source: test, new[] {WellKnownMetadataReferences.NonBlockingConcurrentDictionary});
+        }
+
+        [Fact]
+        public Task AssertTrueForCreationOfNewInstanceWithMessageIsBannedAsync()
+        {
+            const string test = @"
+     using System.Collections.Concurrent;
+
+     namespace ConsoleApplication1
+     {
+         class TypeName
+         {
+             void Test()
+             {
+                 ConcurrentDictionary<int,int> dictionary = new ConcurrentDictionary<int,int>();
+             }
+         }
+     }";
+            DiagnosticResult expected = new()
+                                        {
+                                            Id = "FFS0031",
+                                            Message = @"Use NonBlocking.ConcurrentDictionary  rather than System.Collections.Concurrent.ConcurrentDictionary",
+                                            Severity = DiagnosticSeverity.Error,
+                                            Locations = new[] {new DiagnosticResultLocation(path: "Test0.cs", line: 10, column: 61)}
+                                        };
+
+            return this.VerifyCSharpDiagnosticAsync(source: test, new[] {WellKnownMetadataReferences.ConcurrentDictionary}, expected);
+        }
+
+        [Fact]
+        public Task AssertTrueForParameterInMethodDeclarationWithMessageIsBannedAsync()
+        {
+            const string test = @"
+     using System.Collections.Concurrent;
+
+     namespace ConsoleApplication1
+     {
+         class TypeName
+         {
+             void Test(ConcurrentDictionary<int,int> dictionary)
```

Comment:
Does this cope with constructor parameters too?

Reply:
Test for constructor parameter added.